app/fissile.go

```diff
@@ -791,77 +791,95 @@ func (f *Fissile) generateAuth(settings kube.ExportSettings) error {
 		return err
 	}
 
-	// Check the accounts for the pod security policies they
-	// reference, and create their cluster roles. The necessary
-	// cluster role bindings will be created by NewRBACAccount.
-
-	for _, pspName := range model.PodSecurityPolicies() {
-		for _, accountSpec := range settings.RoleManifest.Configuration.Authorization.Accounts {
-			if accountSpec.PodSecurityPolicy == pspName {
-				node, err := kube.NewRBACClusterRolePSP(pspName, settings)
-				if err != nil {
-					return err
-				}
-				err = f.writeHelmNode(authDir, fmt.Sprintf("auth-clusterrole-%s.yaml", pspName), node)
-				if err != nil {
-					return err
-				}
-				break
-			}
+	// Generate accounts (and any associated role bindings / cluster role bindings)
+	for accountName, accountSpec := range settings.RoleManifest.Configuration.Authorization.Accounts {
+		// Ignore accounts referenced by a single instance group. These are not
+		// written as their own files, but as part of the instance group.
+		if len(accountSpec.UsedBy) < 2 {
+			continue
+		}
+
+		nodes, err := kube.NewRBACAccount(accountName, settings.RoleManifest.Configuration, settings)
+		if err != nil {
+			return err
+		}
+		err = f.writeHelmNode(authDir, fmt.Sprintf("account-%s.yaml", accountName), nodes...)
+		if err != nil {
+			return err
 		}
 	}
 
+	// Generate roles
 	for roleName, roleSpec := range settings.RoleManifest.Configuration.Authorization.Roles {
-		// Ignore roles referenced by a single instance
-		// group. These are not written as their own files,
-		// but as part of the instance group.
-		if settings.RoleManifest.Configuration.Authorization.RoleUse[roleName] < 2 {
+		var accountNames []string
+		for accountName := range settings.RoleManifest.Configuration.Authorization.RoleUsedBy[roleName] {
+			accountNames = append(accountNames, fmt.Sprintf("- %s", accountName))
+		}
+		if len(accountNames) < 1 {
+			panic(fmt.Sprintf("Role %s used by no accounts", roleName))
+		}
+		if len(accountNames) < 2 {
+			// Ignore roles referenced by a single account. These are not written as their own files,
+			// but as part of the account.
 			continue
 		}
+		sort.Strings(accountNames)
 
-		node, err := kube.NewRBACRole(roleName, roleSpec, settings)
+		node, err := kube.NewRBACRole(roleName, kube.RBACRoleKindRole, roleSpec, settings)
 		if err != nil {
 			return err
 		}
+		node.Set(helm.Comment(fmt.Sprintf(`Role %s used by accounts:\n%s`, roleName, strings.Join(accountNames, "\n"))))
```

Comment:
Quoting issue: The `backtics` mean that the `\n` is printed as-is, instead of as newline. Use double-quotes instead. See also the comment later for related thoughts.

Reply:
Fixed in 12d0ca53.